cohesity/resource_cohesity_physical_edition_cluster.go

```diff
@@ -215,22 +217,40 @@ func resourceCohesityPhysicalEditionClusterCreate(resourceData *schema.ResourceD
 		EnableFipsMode:   &enableFipsMode,
 		RotationPeriod:   &encryptionKeysRotationPeriod,
 	}
-	requestBody.NetworkConfig = models.NetworkConfiguration{
-		ClusterGateway:    &clusterGateway,
+
+	networkConfig := models.NetworkConfiguration{
 		ClusterSubnetMask: &clusterSubnetMask,
-		DnsServers:        &dnsServers,
-		DomainNames:       &domainNames,
 		NtpServers:        &ntpServers,
 		VipHostname:       &vipHostName,
-		Vips:              &vips,
+		DnsServers:        &dnsServers,
 	}
+	if len(domainNames) != 0 {
+		networkConfig.DomainNames = &domainNames
+	}
+	if len(vips) != 0 {
+		networkConfig.Vips = &vips
+	}
+	if clusterGateway != "" {
+		networkConfig.ClusterGateway = &clusterGateway
+	}
+	requestBody.NetworkConfig = networkConfig
 
-	requestBody.IpmiConfig = models.IpmiConfiguration{
-		IpmiGateway:    &ipmiGateway,
-		IpmiPassword:   &ipmiPassword,
-		IpmiSubnetMask: &ipmiSubnetMask,
-		IpmiUsername:   &ipmiUsername,
+	ipmiConfig := models.IpmiConfiguration{}
+	if ipmiGateway != "" {
```

Comment:
We need to handle a case where users passes this as null object ?

Reply:
when no value is assigned for string variable, it takes empty string by default, this is handled